app/src/org/commcare/android/view/EntityView.java

```diff
@@ -530,5 +491,8 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
                 views[i].setLayoutParams(params);
             }
         }
-    }*/
+        
+        // Use children's new widths
+        super.onMeasure(widthMeasureSpec, heightMeasureSpec);

```

Comment:
@phillipm I doubt it in this case.

This setup is relying on the layout engine to still be performing sizing on the height of all of the associated views, so we can't set anything manually without doing those calculations on our own.

The first call to onMeasure results in this view being clear on its height and width, after which it sets the layoutparams for its children to force them to a certain width. After it does that, this view needs both the children and itself to have heights recalculated, which you can only trigger by asking for a recursive measurement again.